### DISCUSSION TITLE
1GB/1vcpu server enough?

### DISCUSSION THREAD

**Discussion by test2a** [rustdesk/rustdesk#4857] [Q&A]
Hi.

I see that rustdesk currently uses 2vCPU/4GB ram VPS servers for public use while some are getting shut down for some reason. Anyway, my question is, has anyone tested the smallest vps and how much traffic/users can such a vps handle?

i'd be willing to buy and set up my own server or 2 for redundancy so how can that be achieved? the redundancy i mean.

second, do we need login functionality for own servers or they can be unlisted but without login? like i will have rd.adg.com and i am thinking, i will ask my friends to install RD and use this server in this and this place. do they need to login or create an account first and then it will work?

**Comment by SergeyMy** [comment]
everything will work at once, **for ALL** without authorization.
encryption only for those who have the public key

**Comment by test2a** [comment]
@SergeyMy hey. if we use docker, whats the usage like? ram or cpu usage for example? i have a small vps like i said and i would like to use that for rustdesk and my other work

**Comment by SergeyMy** [comment]
docker is used to isolate the project from the entire system, or if you do not want to build the software yourself.

hbbs hbbr services are overkill! I just put this together
1core 2Gb-RAM
4.1.13-100.fc21.i686+PAE 
:):):):)

**Comment by DevJupyHUB** [comment]
@test2a "question is, has anyone tested the smallest vps and how much traffic/users can such a vps handle?"

Hi there,

This configuration works without problems, with 150 - 200 clients...
you can see the memory usage (400-500MB / 2vCPU AWS t3-micro)), and it's also running: CF argo tunnel, Crowdsec firewall bouncer + Filebeat to ELK

![image](https://github.com/rustdesk/rustdesk/assets/125738232/a7d1fd36-d664-4e81-8761-2bc1147f8c19)

BR,
SedonD

**Reply by test2a** [reply]
@DevJupyHUB so according to this https://rustdesk.com/docs/en/self-host/install/#step-2--run-hbbs-and-hbbr-on-your-server

you just download hbbr and hbbs and that's it? 

how would you set up hostname , the docs aren;t clear on it

**Reply by shvchk** [reply]
@test2a rustdesk-server memory usage is ~7-11 MB total (hbbs + hbbr + everything else in the container) as reported by `htop`, see RES and [PSS (proportional set size)](https://en.wikipedia.org/wiki/Proportional_set_size) columns:

```
 RES   PSS PSSWP CPU% Command
2072   806     0  0.0 │  └─ /usr/bin/conmon --api-version 1 -c AAA -u BBB
  64    56     0  0.0 │     └─ /package/admin/s6/command/s6-svscan -d4 -- /run/service
  64    24     0  0.0 │        ├─ s6-supervise s6-linux-init-shutdownd
   4    44     0  0.0 │        │  └─ /package/admin/s6-linux-init/command/s6-linux-init-shutdownd -c /run/s6/basedir -g 3000 -C -B
  68    28     0  0.0 │        ├─ s6-supervise s6rc-fdholder
  64    24     0  0.0 │        ├─ s6-supervise s6rc-oneshot-runner
   4    40     0  0.0 │        │  └─ /package/admin/s6/command/s6-ipcserverd -1 -- /package/admin/s6/command/s6-ipcserver-access -v0 -E -l0 -i data/rules -- /package/admin/s6/comman
  64    24     0  0.0 │        ├─ s6-supervise hbbr
 404  1250     0  0.0 │        │  └─ sh ./run
 700  2654     0  0.0 │        │     └─ /usr/bin/hbbr -k _
  64    24     0  0.0 │        └─ s6-supervise hbbs
 404  1394     0  0.0 │           └─ sh ./run
2892  5298     0  0.0 │              └─ /usr/bin/hbbs -r example.com -k _
```

It was rolled out with a single command* ([see readme](https://github.com/rustdesk/rustdesk-server#s6-overlay-based-images)):

```sh
podman run --name rustdesk-server \
  --net host \
  -e "RELAY=example.com" \
  -e "ENCRYPTED_ONLY=1" \
  -v "/srv/rustdesk-server/data:/data" \
  -d docker.io/rustdesk/rustdesk-server-s6
```

As you see, hostname is set with a `RELAY` env var. I'm using rootless podman container instead of docker here.

\* ok, you probably want to `mkdir -p /srv/rustdesk-server/data` first, and open fw ports, but that's basically it.

**Comment by DevJupyHUB** [comment]
_"yes, the documentation still needs improvement, but used together with github it can be adapted, sometimes the source code helps"_

use this, it's great and you'll forget all your headaches :-):
https://github.com/techahold/rustdeskinstall


+++edit:
the "script"  - also perfectly suitable for update

wget https://raw.githubusercontent.com/techahold/rustdeskinstall/master/update.sh
chmod +x update.sh
./update.sh

**Comment by lucas-0liveira** [comment]
It's a good idea to test RustDesk on the smallest VPS to see how it performs. Usually, smaller servers might have limitations in handling concurrent users or high traffic, so it's important to monitor its performance. Setting up redundancy involves having a backup server in case the primary one goes down. This can be achieved through various methods such as load balancing or having a failover setup. If you're using a cloud provider like AWS or Google Cloud, they often have services to help with redundancy.

**Comment by DevJupyHUB** [comment]
@lucas-0liveira "_good idea to test RustDesk on the smallest VPS to see how it performs_  - AWS or Google Cloud, they often have services to help with redundancy"

_As you can read above, it performs quite well... :)_

You mention redundancy, - do not think about anything serious, because you will not get a good value/cost ratio on a small VPS like AWS t2 or t3, - on the AWS just keep a well-functioning backup "policy - EBS snapshot =Cost ($) **0,02/day**" and even with resizing EC2 or not, you can restore everything in a minute...

![image](https://github.com/rustdesk/rustdesk/assets/125738232/c802227c-b1c1-4f10-a6fc-5614e8742ce8)

![image](https://github.com/rustdesk/rustdesk/assets/125738232/1f860cc2-ea0e-4f97-b2f7-60e3be9e3704)
